src/SoftekBarcodeAPI.php

```diff
@@ -37,8 +40,101 @@ public function __construct(SoftekBarcodeConfig $configuration)
 
             // license the instance for further operations
             $this->ffi->mtSetLicenseKey($this->instance, $configuration->getLicenseKey());
-        } catch (\Throwable $e) {
-            throw new SoftekInitializationException($e->getMessage(), "", $e);
+        } catch (Throwable $e) {
+            throw new SoftekInitializationException($e->getMessage(), 0, $e);
+        }
+    }
+
+    /**
+     * Scan the specified image file for bar code strings and return the detected barcodes in the file as Objects.
+     * Note that the function will stop when the first barcode is found in a document unless
+     * the MultipleRead property is set to True
+     * https://www.bardecode.com/en1/scanbarcode-scan-an-image-for-barcodes/
+     *
+     * @param string $pathToFile
+     * @return BarcodeScanResult[]
+     * @throws SoftekProcessImageException if an error occurs during extraction of the barcode from the InputFile
+     */
+    public function processImage(string $pathToFile): array
```

Comment:
Is it possible to call `processImage` multiple times on different images with this implementation? If it is not possible or requires calling some `reset` function on the FFI, we should look into that.

Reply:
i will find out what we have an functions to reset processes

Reply:
i do not find anythin to an clearing function or reset function.